knowledge_repo/app/models.py

```diff
@@ -350,8 +350,18 @@ def groups(self, groups):
 
         # create an implicit group, group_post.id, to add
         # single users to
-        group = Group(name="group_" + self.id)
+        # we don't have self.id here necessarily, so query the Post table
```

Comment:
I tested this in two ways, the first was by deleting my original db so that all posts would be re-added. In that case, there was no `self.id`, which was interesting to me since we actually add the post. From this post: http://stackoverflow.com/questions/23076483/sqlalchemy-get-added-objects-id-without-committing-session, it looks like we may need to flush the object, so I can try that to see if we can just use the id, since that's far less brittle